public/app/plugins/panel/trend/TrendPanel.tsx

```diff
@@ -39,7 +39,7 @@ export const TrendPanel = ({
     dimFields = {
       ...dimFields,
       x: (field, frame, frames) => {
-        return getFieldDisplayName(field, frame, frames) === trendXFieldName;
+        return field.name === trendXFieldName;
```

Comment:
rather than trying to do the calculation ourselves, should we just use a matcher?

I think this could be:
```suggestion
        x: fieldMatchers.get(FieldMatcherID.byName).get(trendXFieldName),
```